pkg/ingester/ingester_test.go

```diff
@@ -5757,6 +5757,126 @@ func TestGetIgnoreSeriesLimitForMetricNamesMap(t *testing.T) {
 	require.Equal(t, map[string]struct{}{"foo": {}, "bar": {}}, cfg.getIgnoreSeriesLimitForMetricNamesMap())
 }
 
+func Test_Ingester_QueryOutOfOrder(t *testing.T) {
```

Comment:
Can we also add a (sub?)test for changing OOOTW in runtime?

Reply:
(the mimir-prometheus cases in question are [here](https://github.com/grafana/mimir-prometheus/blob/df59320886e03a555d379ac4b0b3130f661407e0/tsdb/db_test.go#L5019-L5169))

Reply:
Added the test in https://github.com/grafana/mimir/pull/2187/commits/afdefc6563b0aee168eeb193320111ddc49fbd85

Reply:
I have replaced the above commit with https://github.com/grafana/mimir/pull/2187/commits/257d2342ed7143b4414ea42b550331813c03719d. The diff looks harder to read in the commit. It will be easier to review by looking at the diff in https://github.com/grafana/mimir/pull/2187/files